src/py_sonar_scanner/__main__.py

```diff
@@ -17,19 +17,29 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
+import sys
 
 from py_sonar_scanner.configuration import Configuration
 from py_sonar_scanner.environment import Environment
+from py_sonar_scanner.logger import ApplicationLogger
 
 
 def scan():
-    cfg = Configuration()
-    cfg.setup()
+    log = ApplicationLogger.get_logger()
+    debug_enabled = False
+    try:
+        cfg = Configuration()
+        cfg.setup()
+        debug_enabled = cfg.is_debug()
 
-    env = Environment(cfg)
-    env.setup()
-    env.scanner().scan()
-    env.cleanup()
+        env = Environment(cfg)
+        env.setup()
+        env.scanner().scan()
+        env.cleanup()
+    except Exception as e:
+        log.exception("Error during SonarScanner execution: %s", str(e))
+        if not debug_enabled:
```

Comment:
Minor suggestion, feel free to disregard:
I guess we could just inline `cfg.is_debug()` call here, right?
Though I guess it was not really nice to do since `cfg.setup()` could raise exceptions. After you rebase, it probably won't be the case anymore, then we could move the instantiation and setup of `Configuration` outside of the `try`, wdyt?

edit: comment was written before the rebase, but I think the refactoring is indeed possible.